lib/vm/src/vmcontext.rs

```diff
@@ -15,15 +15,37 @@ use std::ptr::{self, NonNull};
 use std::sync::Arc;
 use std::u32;
 
+/// We stop lying about what this daat is
+/// TODO:
+#[derive(Copy, Clone)]
+pub union VMFunctionExtraData {
```

Comment:
Optional: consider the name VMFunctionContextParameter or VMFunctionEnvParameter, or possibly CallbackParameter. Whether it's a wasm function or a host function this is giving that function its context/env.

Reply:
I'm down for renaming, but I'll wait for more feedback before renaming!  I like your suggestions though

Reply:
I also like `VMFunctionContextParameter`. I would have use `VMFunctionEnvironment` first, but it creates a naming conflict with “host environment” or similar.

Reply:
Don't forget to rename the `extra_data` field everywhere accordingly too :-).

Reply:
I like `VMFunctionEnvironment`